WooCommerce/Classes/ViewRelated/Products/Variations/Bulk Update/BulkUpdateViewModel.swift

```diff
@@ -59,26 +76,99 @@ final class BulkUpdateViewModel {
         let pageNumber = Constants.pageNumber
         let action = ProductVariationAction
             .synchronizeProductVariations(siteID: siteID, productID: productID, pageNumber: pageNumber, pageSize: numberOfObjects) { [weak self] error in
-                guard let self = self else {
-                    return
-                }
+                guard let self = self else { return }
 
                 if let error = error {
                     self.syncState = .error
 
                     DDLogError("⛔️ Error synchronizing product variations: \(error)")
                 } else {
-                    self.syncState = .synced
+                    self.configureResultsControllerAndFetchData { [weak self] error in
+                        if error != nil {
+                            self?.syncState = .error
+                        } else {
+                            self?.syncState = .synced
+                        }
+                    }
                 }
             }
 
         storesManager.dispatch(action)
     }
+
+    /// Provides the view model with all user facing data of the option for updating the regular price
+    ///
+    func viewModelForDisplayingRegularPrice() -> ValueOneTableViewCell.ViewModel {
+        return ValueOneTableViewCell.ViewModel(text: Localization.updateRegularPriceTitle,
+                                               detailText: detailTextforPrice(\ProductVariation.regularPrice))
+    }
+
+    /// Returns the detail text of the option for updating a price
+    ///
+    private func detailTextforPrice(_ priceKeyPath: KeyPath<ProductVariation, String?>) -> String {
+        guard let bulkUpdateFormModel = bulkUpdateFormModel else {
+            return ""
+        }
+        switch bulkUpdateFormModel.bulkValueOf(priceKeyPath) {
+        case .none:
+            return Localization.noneValue
+        case .mixed:
+            return Localization.mixedValue
+        case let .value(price):
+            return formatPriceString(price)
+        }
+    }
+
+    /// It formats a price `String` according to the current price settings.
+    ///
+    private func formatPriceString(_ price: String) -> String {
+        let currencyCode = currencySettings.currencyCode
+        let currency = currencySettings.symbol(from: currencyCode)
+
+        return currencyFormatter.formatAmount(price, with: currency) ?? ""
+    }
+
+    /// Setup the results controller
+    ///
+    private func configureResultsControllerAndFetchData(onCompletion: @escaping (Error?) -> Void) {
+        resultsController.onDidChangeContent = { [weak self] in
+            guard let self = self else { return }
+            self.updateDataModel()
+            onCompletion(nil)
```

Comment:
Why do we need to call `onCompletion` again here?

Reply:
Since the data changed (e.g. after the bulk update), I need to set the state again to `synced` triggering the UI to display the new data.

Reply:
Hmm, I think this is not ideal as it is quite an implicit behavior. In reality, the state hasn't changed. `.synced -> synced`
Let's leave it like this so we can move forward tho!

Reply:
I really like your idea to have the section data with the state, it is more natural and it will make the communcation VC <-> VM more straightforward since the controller after getting notified for the state change it will not have to ask from the VM the section data.

Updated the Sync State to:

   ```Swift
enum SyncState: Equatable {
        case syncing
        case synced([Section])
        case error
        case notStarted
    }
```